web/src/hashchange.js

```diff
@@ -65,6 +65,21 @@ function show_all_message_view() {
     });
 }
 
+function require_hash_change(hash) {
+    const terms = hash_util.parse_narrow(hash);
```

Comment:
We already called `hash_util.parse_narrow(hash);`; should this be passed from the caller?